python/pacemaker/_cts/watcher.py

```diff
@@ -229,33 +231,33 @@ def async_complete(self, pid, returncode, out, err):
         out         -- stdout from the journal read
         err         -- stderr from the journal read
         """
-        found_cursor = False
-        for line in out:
-            match = re.search(r"^-- cursor: ([^.]+)", line)
+        parser = isoparser()
+        dt_limit = None
+        if self.limit:
+            dt_limit = parser.isoparse(self.limit)
 
+        for line_num, line in enumerate(out):
+            match = re.search(r"^-- cursor: ([^.]+)", line)
             if match:
-                found_cursor = True
                 self.offset = match.group(1).strip()
-                self.debug("Got %d lines, new cursor: %s" % (len(out), self.offset))
-            else:
-                self.cache.append(line)
+                self.debug("Got %d lines before new cursor: %s"
+                           % (line_num, self.offset))
+                break
 
-        if self.limit and not found_cursor:
-            self._hit_limit = True
-            self.debug("Got %d lines but no cursor: %s" % (len(out), self.offset))
+            if dt_limit:
+                match = re.search(r"^\S+", line)
+                if not match:
+                    continue
 
-            # Get the current cursor
-            # pylint: disable=not-callable
-            (_, out) = self.rsh(self.host, "journalctl -q -n 0 --show-cursor", verbose=0)
```

Comment:
Based on a9c45ec0, it looks like the purpose of this section was to try to make it safe to run multiple journal log captures *from a single host* at one time.
* I don't believe we actually do that currently.
* If we were to do it, we'd want to ensure that the threads' outputs were placed into the line cache in the correct order when the threads are joined.

We can get rid of the added complexity for now and make this more robustly thread-safe later if we decide to use it that way.